modules/plugin/geopkg/src/main/java/org/geotools/geopkg/GeoPkgDialect.java

```diff
@@ -688,9 +707,15 @@ public void setValue(
                 ps.setString(column, value.toString());
                 break;
             case Types.TIMESTAMP:
-                // 2020-02-19 23:00:00.0  --> 2020-02-19 23:00:00.0Z
+                // 2020-02-19 23:00:00.0  --> 2020-02-19T23:00:00.0Z
                 // We need the "Z" or sql lite will interpret the value as local time
-                ps.setString(column, value.toString() + "Z");
+                // geopkg - format will be ISO-8601 - YYYY-MM-DDTHH:MM[:SS.SSS]Z
+                if (value != null) {
+                    var v = ((Timestamp) value).toInstant().toString();
+                    ps.setString(column, v);
+                } else {
+                    ps.setString(column, null);
+                }
```

Comment:
Unnecessary, the code at the beginning of the method is already checking for null, using ``setNull``, and exiting.